R/sample_grids_proposed.R

```diff
@@ -0,0 +1,177 @@
+  }
+
+  ## create 'Priority' field
+  surv_cent$Priority <- "'Survey or Extra'"
+
+  ## create X Y coordinate columns from geometry
+  surv_cent$X <- sf::st_coordinates(surv_cent)[,1]
+  surv_cent$Y <- sf::st_coordinates(surv_cent)[,2]
+
+
+  ## basin specific shape files and bounding box
+  if(basin == "WB"){
+  bound_box <- c(xmin = -83.550, ymin = 41.3494, xmax = -82.450, ymax = 42.1053)
+  bounds <- bound_box %>% sf::st_bbox() %>% sf::st_as_sfc() %>% sf::st_set_crs(4326)
+  basin_shape <- eriespatial::shape_wbstrata
+  } else if(basin == "CB"){
+  bound_box <- c(xmin = -82.4, ymin = 41.363, xmax = -80.4, ymax = 42.7205)
+  bounds <- bound_box %>% sf::st_bbox() %>% sf::st_as_sfc() %>% sf::st_set_crs(4326)
+  basin_shape <- eriespatial::shape_cbstrata
+  } else if(basin == "EB"){
+  bound_box <- c(xmin = -80.5, ymin = 42.1, xmax = -78.85, ymax = 42.9)
+  bounds <- bound_box %>% sf::st_bbox() %>% sf::st_as_sfc() %>% sf::st_set_crs(4326)
+  basin_shape <- eriespatial::shape_ebstrata
+  } else {base::print("Check basin name")}
+
+  ##  plot all available centroids
+  p1 <- eriespatial::base_erieshore +
+    ggplot2::scale_x_continuous(limits = c(bound_box["xmin"], bound_box["xmax"]))+
+    ggplot2::scale_y_continuous(limits = c(bound_box["ymin"], bound_box["ymax"]))+
+    ggplot2::geom_sf(data = shape_5mingrid_surv_sub, col="lightgray", fill=NA, lwd = 0.5, alpha = 0.5) +
+    ggplot2::geom_sf(data = basin_shape, aes(fill=STRATUM, alpha=0.5)) +
+    ggplot2::scale_fill_viridis_d(alpha = 0.5) +
+    ggplot2::guides(alpha="none") +
+    ggplot2::geom_sf(data = shape_5mincent_surv_sub, color = "black", pch = 16) +
+    ggplot2::geom_text(data = shape_5mincent_surv_sub, aes(label = GRID, x = X, y = Y),
+              nudge_x = 0, nudge_y = 0.025, size = 2.0) +
+    ggplot2::ylab("Latitude (dd)") + xlab("Longitude (dd)") +
+    ggplot2::labs(title = "Available sample grids") +
+    ggplot2::theme_bw() +
+    ggplot2::theme(legend.position = "bottom")
+
+  ## plot randomly selected centroids
+  p2 <- eriespatial::base_erieshore +
+    ggplot2::scale_x_continuous(limits = c(bound_box["xmin"], bound_box["xmax"]))+
+    ggplot2::scale_y_continuous(limits = c(bound_box["ymin"], bound_box["ymax"]))+
+    ggplot2::geom_sf(data = shape_5mingrid_surv_sub, col="lightgray", fill=NA, lwd = 0.5, alpha = 0.5) +
+    ggplot2::geom_sf(data = basin_shape, aes(fill=STRATUM, alpha=0.5)) +
+    ggplot2::scale_fill_viridis_d(alpha = 0.5) +
+    ggplot2::guides(alpha="none") +
+    ggplot2::geom_sf(data = surv_cent, color = "black", pch = 16) +
+    ggplot2::geom_text(data = surv_cent, aes(label = GRID, x = X, y = Y),
+              nudge_x = 0, nudge_y = 0.025, size = 2.5) +
+    ggplot2::ylab("Latitude (dd)") + xlab("Longitude (dd)") +
+    ggplot2::labs(title = "Randomly selected sample grids") +
+    ggplot2::theme_bw() +
+    ggplot2::theme(legend.position = "bottom")
+
+  ## combine two plots into one
+  p12 <<- gridExtra::grid.arrange(p1, p2, ncol = 2)
```

Comment:
is this an intentional global assignment?

Reply:
It was at the time of testing, but I think we can remove it now.

Reply:
removed here: [c842933](https://github.com/LakeErie-PreyFish-Acoustics/erieacoustics/pull/47/commits/c84293365ed061414fbfce5b09d39090db42c261)